fmskill/data/altimetry.py

```diff
@@ -0,0 +1,754 @@
+        area : str
+            area specification in one of three allowed formats:
+                - polygon=6.811,54.993,8.009,54.993,8.009,57.154,6.811,57.154,6.811,54.993
+                - bbox=115,28,150,52
+                - lon=10.9&lat=55.9&radius=100
+        start_time : str or datetime, optional
+            start of time interval, by default "2020-01-01"
+        end_time : str or datetime, optional
+            end of time interval, by default datetime.now()
+        satellites : str, optional
+            Satellites to be downloaded, e.g. '', '3a', 'j3, by default '' (=all)
+
+        Returns
+        -------
+        pd.DataFrame
+            number of observations per day
+        """
+        url = self.API_URL + "temporal-coverage"
+        payload = self._area_time_sat_payload(area, start_time, end_time, satellites)
+        r = requests.get(url, params=payload, headers=self.HEADERS)
+        if r.status_code != 200:
+            print(r.text)
+        r.raise_for_status()
+        data = r.json()
+        df = pd.DataFrame(data["temporal_coverage"])
+        df["date"] = pd.to_datetime(df["date"])
+        return df.set_index("date")
+
+    def get_spatial_coverage(
+        self, area, start_time="20200101", end_time=None, satellites=""
+    ):
+        """Get spatial observation coverage as count per spatial bin in a
+        rectangle covering the specified area
+
+        Parameters
+        ----------
+        area : str
+            area specification in one of three allowed formats:
+                - polygon=6.811,54.993,8.009,54.993,8.009,57.154,6.811,57.154,6.811,54.993
+                - bbox=115,28,150,52
+                - lon=10.9&lat=55.9&radius=100
+        start_time : str or datetime, optional
+            start of time interval, by default "2020-01-01"
+        end_time : str or datetime, optional
+            end of time interval, by default datetime.now()
+        satellites : str, optional
+            Satellites to be downloaded, e.g. '', '3a', 'j3, by default '' (=all)
+
+        Returns
+        -------
+        geopandas.GeoDataFrame
+            count per spatial bin
+        """
+        try:
+            import geopandas as gpd
+        except ImportError:
+            raise ImportError(
+                "The geopandas package is required by the get_spatial_coverage() method. Install it with 'conda install geopandas' or 'pip install geopandas'"
+            )
+
+        url = self.API_URL + "spatial-coverage"
+        payload = self._area_time_sat_payload(area, start_time, end_time, satellites)
+        r = requests.get(url, params=payload, headers=self.HEADERS)
+        if r.status_code != 200:
+            print(r.text)
+        r.raise_for_status()
+        data = r.json()
+        if data and "coverage" in data:
+            gdf = gpd.GeoDataFrame.from_features(data["coverage"], crs="epsg:4326")
+            return gdf
+
+    def get_altimetry_data(
+        self,
+        area,
```

Comment:
I think it should be possible to pass in a dfsu/mesh/domain object as an argument to the area.

Because of the limitation in the current API implementation, it is impossible to pass in a complicated polygon as a parameter.

1. Extract simple polygon from dfsu object, e.g convex hull, e.g. `mr.dfs.to_shapely().buffer(0).convex_hull`
2. Get data from API with coarse filter from simple polygon
3. Filter data in Python with original polygon e.g `gdf_filtered = gdf[gdf.geometry.within(aoi)]`

Reply:
Totally agree, see my comment above - I just would like to postpone it to another PR.